external-provider-ui/src/main/java/org/jahia/modules/external/admin/Activator.java

```diff
@@ -14,13 +16,20 @@
 @Component(immediate = true)
 public class Activator {
 
+    private static final Logger logger = LoggerFactory.getLogger(Activator.class);
+
     @Activate
     public void start(BundleContext context) {
         Set<Bundle> wiredBundles = context.getBundle().adapt(BundleWiring.class).getProvidedWires("osgi.wiring.package").stream().map(x -> x.getRequirer().getBundle()).collect(Collectors.toSet());
         Set<Bundle> toRefresh = Arrays.stream(context.getBundles())
                 .filter(b -> b.getHeaders().get("Import-Package") != null && b.getHeaders().get("Import-Package").contains("org.jahia.modules.external.admin"))
                 .filter(b -> !wiredBundles.contains(b))
+                .filter(b -> b.getState() == Bundle.ACTIVE)
```

Comment:
we need to refresh also if state is RESOLVED

Reply:
Added check for RESOLVED also, test seems ok (no hanging)